packages/vitest-plugin-storybook/src/global-setup.ts

```diff
@@ -0,0 +1,81 @@
+import { spawn, type ChildProcess } from 'node:child_process'
+import { log } from './utils'
+import http from 'node:http'
+import https from 'node:https'
+
+import type { GlobalSetupContext } from 'vitest/node'
+
+let storybookProcess: ChildProcess | null = null
+
+const checkStorybookRunning = (storybookUrl: string) => {
+  return new Promise<boolean>((resolve) => {
+    const url = new URL(`${storybookUrl}/iframe.html`)
+    const protocol = url.protocol === 'https:' ? https : http
+
+    const options = {
+      method: 'HEAD',
+      host: url.hostname,
+      path: url.pathname,
+    }
+
+    const req = protocol.request(options, (res) => {
+      resolve(res.statusCode === 200)
+    })
+
+    req.on('error', () => resolve(false))
+    req.end()
+  })
+}
+
+const startStorybookIfNeeded = async () => {
+  const storybookScript = process.env.__STORYBOOK_SCRIPT__ || ''
+  const storybookUrl = process.env.__STORYBOOK_URL__ || ''
+
+  const isRunning = await checkStorybookRunning(storybookUrl)
+
+  if (isRunning) {
+    log('Storybook is already running')
+    return
+  }
+
+  log(`Starting Storybook with command: ${storybookScript}`)
+
+  try {
+    // We do this sync because we don't want Vitest to hang while Storybook is starting
+    storybookProcess = spawn(storybookScript, [], {
+      stdio: 'ignore',
+      cwd: process.cwd(),
+      shell: true,
+    })
+
+    storybookProcess.on('error', (error) => {
+      log('Failed to start Storybook:', error)
+      console.log({ error })
```

Comment:
```suggestion
```
I think logging then rethrowing will cause a double log because of the outer try/catch which also logs. So maybe not log here. And of course no console.log.